File: Serene1.Web/Modules/MovieDB/Genre/GenreForm.cs
using Serenity.ComponentModel;
using System;

namespace Serene1.MovieDB.Forms;

[FormScript("MovieDB.Genre")]
[BasedOnRow(typeof(GenreRow), CheckNames = true)]
public class GenreForm
{
    public string Title { get; set; }
    public string Description { get; set; }
    public string Storyline { get; set; }
    public int Year { get; set; }
    public DateTime ReleaseDate { get; set; }
    public int Runtime { get; set; }
    public int Kind { get; set; }
}

File: Serene1.Web/Modules/ServerTypes/MovieDB/GenreForm.ts
import { StringEditor, IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface GenreForm {
    Title: StringEditor;
    Description: StringEditor;
    Storyline: StringEditor;
    Year: IntegerEditor;
    ReleaseDate: DateEditor;
    Runtime: IntegerEditor;
    Kind: IntegerEditor;
}

export class GenreForm extends PrefixedContext {
    static formKey = 'MovieDB.Genre';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!GenreForm.init)  {
            GenreForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = DateEditor;

            initFormType(GenreForm, [
                'Title', w0,
                'Description', w0,
                'Storyline', w0,
                'Year', w1,
                'ReleaseDate', w2,
                'Runtime', w1,
                'Kind', w1
            ]);
        }
    }
}